src/cache.cc

```diff
@@ -205,8 +205,10 @@ bool CACHE::handle_miss(const PACKET& handle_pkt)
 
     if (handle_pkt.fill_this_level)
       fwd_pkt.to_return = {this};
-    else
+    else {
+      fwd_pkt.fill_this_level = true; // We will fill the next level
       fwd_pkt.to_return.clear();
+    }
 
     fwd_pkt.prefetch_from_this = false;
```

Comment:
```suggestion
    else
      fwd_pkt.to_return.clear();
    
    fwd_pkt.fill_this_level = true; // We will always fill the lower level
    fwd_pkt.prefetch_from_this = false;
```

This is minor, but I think moving the assignment down to match the assignment of `prefetch_from_this` expresses the intent better. It does the same thing, but more clearly says "We will always fill the lower level".